pkg/client/apiutil/restmapper.go

```diff
@@ -162,25 +168,34 @@ func (m *mapper) addKnownGroupAndReload(groupName string, versions ...string) er
 	// If no specific versions are set by user, we will scan all available ones for the API group.
 	// This operation requires 2 requests: /api and /apis, but only once. For all subsequent calls
 	// this data will be taken from cache.
-	if len(versions) == 0 {
-		apiGroup, err := m.findAPIGroupByName(groupName)
+	//
+	// Calling this also tells us if the server supports aggregated discovery, which allows
+	// loading everything with two api calls which we assume is overall cheaper, hence we always
+	// do this once optimistically to check and keep doing it if its supported.
```

Comment:
How is the "keep doing it if its supported" part implemented?

As far as I can tell we just always try it by calling GroupsAndMaybeResources, independent of if it's supported or not (which I think is correct, just a bit confused by the comment)

Reply:
That comment is outdated actually, I originally made it always call only this if supported but later realized that for the reloading case, we know the GV and directly fetching that is cheaper as its only one api call vs two here. Updated the go docs accordingly.

Reply:
Thx! That clears up a lot of confusion on my side :)